src/Extensions/LeftAndMainSubsites.php

```diff
@@ -228,18 +223,55 @@ public function canAccess()
             return true;
         }
 
-        // Check if we have access to current section on the current subsite.
-        $accessibleSites = $this->owner->sectionSites(true, 'Main site', $member);
-        return $accessibleSites->count() && $accessibleSites->find('ID', SubsiteState::singleton()->getSubsiteId());
+        // Check we have a member
+        if (!$member) {
+            return null;
+        }
+
+        // Check that some subsites exist first
+        if (!Subsite::get()->exists()) {
+            return null;
+        }
+
+        // Get the current subsite ID
+        $currentSubsiteId = SubsiteState::singleton()->getSubsiteId();
+
+        // Check against the current user's associated groups
+        $allowedInSubsite = false;
+        foreach ($member->Groups() as $group) {
+            // If any of the current user's groups have been given explicit access to the subsite, delegate to core
+            if ($group->AccessAllSubsites) {
+                $allowedInSubsite = true;
+                break;
+            }
+
+            // Check if any of the current user's groups have been given explicit access to the current subsite
+            $groupSubsiteIds = $group->Subsites()->column('ID');
+            if (in_array($currentSubsiteId, $groupSubsiteIds)) {
+                $allowedInSubsite = true;
```

Comment:
Yeah I'd considered that. I think the reason I left it like this is because the two if statements contribute to the same criteria of "is allowed in subsite" so I left them separate for readability. That OK?

Reply:
Sure - I'm just happy the break is in now :)